src/main/java/org/gradle/crypto/checksum/Checksum.java

```diff
@@ -19,25 +19,29 @@
 import com.google.common.hash.HashFunction;
 import com.google.common.hash.Hashing;
 import com.google.common.io.Files;
-import org.gradle.api.Action;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.GradleException;
-import org.gradle.api.file.ConfigurableFileTree;
 import org.gradle.api.file.FileCollection;
-import org.gradle.api.tasks.Input;
-import org.gradle.api.tasks.InputFiles;
-import org.gradle.api.tasks.OutputDirectory;
-import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.file.FileSystemOperations;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.tasks.*;
 import org.gradle.api.tasks.incremental.IncrementalTaskInputs;
-import org.gradle.api.tasks.incremental.InputFileDetails;
 
+import javax.inject.Inject;
 import java.io.File;
 import java.io.IOException;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+@CacheableTask
+public abstract class Checksum extends DefaultTask {
+
+    @Inject abstract public FileSystemOperations getFs();
```

Comment:
One question @eskatos , I want to have this plugin configuration cache compatible, which pass through _service dependency injection_ iiuc.

This is at the price of losing compatibility with Gradle < 5.5 as you confirmed injection was supported by this version.

**Is there an easy alternative for older versions?**
worst case scenario this can be defined as a requirement and older Gradle versions can still rely on lover versions of the plugin.

Reply:
See this discussion for details and answers
https://gradle.slack.com/archives/CDYR952SH/p1644321786538419